scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -0,0 +1,471 @@
+
+    .PARAMETER E2ETestFolder
+        Path of E2E test folder
+
+    .PARAMETER ReleaseLabel
+        Release label, can be uniquely identify the build (e.g <ReleaseName>-<ReleaseAttempt>); which is used as part of Edge device name.
+
+    .PARAMETER ReleaseArtifactImageBuildNumber
+        Release artifact image build number; it is used to construct path of docker images, pulling from EdgeBuilds docker registry. E.g. 20190101.1.
+
+    .PARAMETER Architecture
+        Runtime architecture
+        Note: Valid values are "x64" and "arm32v7".
+
+    .PARAMETER TestName
+        Name of E2E test to be run
+        Note: Valid values are: "TempSensor", "TempFilter", "QuickstartCerts", "TransparentGateway".
+
+    .PARAMETER ContainerRegistryUserName
+        Username of container registry
+
+    .PARAMETER ContainerRegistryPassword
+        Password of given username for container registory
+
+    .PARAMETER IoTHubConnectionString
+        IoT hub connection string for creating edge device
+
+    .PARAMETER EventHubConnectionString
+        Event hub connection string for receive D2C messages
+
+    .EXAMPLE
+        .\Run-E2ETest.ps1 -E2ETestFolder "C:\Data\e2etests" -ReleaseLabel "Release-ARM-1" -ReleaseArtifactImageBuildNumber "20190101.1" -Architecture "x64" -TestName "TempSensor" -ContainerRegistryUsername "EdgeBuilds" -ContainerRegistryPassword "xxxx" -IoTHubConnectionString "xxxx" -EventHubConnectionString "xxxx"
+
+    .NOTES
+        This script is to make running E2E tests easier and centralize E2E test steps in 1 place for reusability.
+        It shares common tasks such as clean up and installation of IoT Edge Security Daemon.
+        Each test should have its own function to implement test steps.
+    #>
+
+[CmdletBinding()]
+Param (
+    [ValidateNotNullOrEmpty()]
+    [ValidateScript({(Test-Path $_ -PathType Container)})]
+    [string] $E2ETestFolder = $(Throw "Path of E2ETest folder is missing or invalid"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $ReleaseLabel = $(Throw "Release label is required"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $ReleaseArtifactImageBuildNumber = $(Throw "Release artifact image build number is required"),
+
+    [ValidateSet("x64", "arm32v7")]
+    [string] $Architecture = $(Throw "Architecture is required"),
+
+    [ValidateSet("TempSensor", "TempFilter", "QuickstartCerts", "TransparentGateway")]
+    [string] $TestName = $(Throw "Test name is required"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $ContainerRegistryUsername = $(Throw "Container registry username is required"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $ContainerRegistryPassword = $(Throw "Container registry password is required"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $IoTHubConnectionString = $(Throw "IoT hub connection string is required"),
+
+    [ValidateNotNullOrEmpty()]
+    [string] $EventHubConnectionString = $(Throw "Event hub connection string is required")
```

Comment:
Can we have default values for some of these? Example: E2ETestfFolder, if not passed, can it the the current folder where you are running the script?  Release Label (If not passed, can we get latest?). Architecture (IF not passed, we could use the architecture of the machine we are running).  TempSensor test could be the default test... The only required field could be IotHubConnectionString (and possibly EventHubConnectionString).

Reply:
Added default values to E2ETestFolder ("." - current folder) and TestName ("All" - running all tests).  Architecture is removed and based on current machine.  For ReleaseLabel, it needs to be unique which is used as part of device id.  Therefore it is still required to provide by users.